profiling/nvtx-connector/kp_nvtx_connector.cpp

```diff
@@ -0,0 +1,199 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <cstdio>
+#include <cstdint>
+#include <vector>
+#include <string>
+
+#include "nvToolsExt.h"
+
+#include "kp_core.hpp"
+
+static int tool_globfences;  // use an integer
+namespace KokkosTools {
+namespace NVProfConnector {
+
+void kokkosp_request_tool_settings(const uint32_t,
+                                   Kokkos_Tools_ToolSettings* settings) {
+  settings->requires_global_fencing = true;
+  if (tool_globfences == 1) {
+    settings->requires_global_fencing = true;
+  } else {
+    settings->requires_global_fencing = false;
+  }
+  // leave the door open for other nonzero values of tools
+}
+
+static uint64_t nextKernelID;
+
+void kokkosp_init_library(const int loadSeq, const uint64_t interfaceVer,
+                          const uint32_t devInfoCount,
+                          Kokkos_Profiling_KokkosPDeviceInfo* deviceInfo) {
+  printf("-----------------------------------------------------------\n");
+  printf("KokkosP: NVTX Analyzer Connector (sequence is %d, version: %llu)\n",
+         loadSeq, interfaceVer);
+  printf("-----------------------------------------------------------\n");
+  const char* tool_global_fences = getenv("KOKKOS_TOOLS_GLOBALFENCES");
+  if (NULL != tool_global_fences) {
+    tool_globfences = atoi(tool_global_fences);
+  } else {
+    tool_globfences =
+        1;  // default to 1 to be conservative for capturing state by the tool
+  }
+
+  char* profileLibrary = getenv("KOKKOS_TOOLS_LIBS");
+  if (NULL == profileLibrary) {
+    printf(
+        "Checking KOKKOS_PROFILE_LIBRARY. WARNING: This is a deprecated "
+        "variable. Please use KOKKOS_TOOLS_LIBS.\n");
+    profileLibrary = getenv("KOKKOS_PROFILE_LIBRARY");
+    if (NULL == profileLibrary) {
+      printf("KokkosP: No library to call in %s\n", profileLibrary);
+      exit(-1);
+    }
```

Comment:
I'm not sure this code is needed...

Reply:
Yeah I don't think for now that this should do this, in particular you don't do anything with it anyway.